lib/internal/main/run.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+/* eslint-disable node-core/prefer-primordials */
+
+// There is no need to add primordials to this file.
+// `run.js` is a script only executed when `node run <script>` is called.
+const {
+  prepareMainThreadExecution,
+  markBootstrapComplete,
+} = require('internal/process/pre_execution');
+const { getPackageJSONScripts } = internalBinding('modules');
+const { execSync } = require('child_process');
+const { resolve, delimiter } = require('path');
+const { escapeShell } = require('internal/shell');
+
+prepareMainThreadExecution(false, false);
+
+markBootstrapComplete();
+
```

Comment:
We need to emit an experimental warning here.

Reply:
The experimental warning is thrown from C++

Reply:
The C++ warning thing just calls back into…emitWarningSync() in JavaScript, which is a lot slower than calling it directly from JavaScript.

Reply:
I guess you will need to reimplement all its logic in C++. But then that includes process warning listener handling, which is in JS.